### PR TITLE
Descriptive stats of generated features

### LINKED ISSUE #78: Descriptive stats of generated features
- [x] Create a list of feature dicts
- [x] Generate a dataset that conforms to that
- [x] Develop from there

Validate the quality of the features based on factors such as 

`pred_polycystic_ovarian_syndrome_within_9999_days_min_fallback_0`

- [x] Source values (e.g. hba1c)
- [x] Lookbehind distance
- [x] Resolve_multiple strategy
- [x] Fallback strategy
- [x] Descriptive stats of values (quartiles, mean, median, (histogram?)
- [x] Proportion that used the fallback strategy
- [x] Unicode histogram: https://github.com/aeturrell/skimpy/blob/3a5025681be9dcd0d6cdeb10bb30d0ad2e052cca/src/skimpy/__init__.py#L271

Next up:
- [x] Figure out what np.nan resolves to
- [x] Run on real data
- [ ] Add to feature generation script
- [x] Test `create_feature_description_from_dir` on Overtaci
- [ ] Fix failing tests

### PR DESCRIPTION
**Pull request by MartinBernstorff**
- [x] I have battle-tested on Overtaci (RMAPPS1279)
- [x] I have bumped the version as described in the wiki

## Notes for reviewers
Some minor chores to skip, but I think you'll be able to spot those :-)

Fixes #78

### CODE REVIEW

File: src/application/t2d/describe_features.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+    MEDICATION_PREDICTORS = create_medication_feature_combinations(
+        LOOKBEHIND_DAYS=LOOKBEHIND_DAYS,
+        RESOLVE_MULTIPLE=["count"],
+        fallback=0,
+    )
+
+    PREDICTOR_LIST = MEDICATION_PREDICTORS + DIAGNOSIS_PREDICTORS + LAB_PREDICTORS
+
+    create_feature_description_from_dir(
```
Comment (HLasse) [inline src/application/t2d/describe_features.py:45]: Add this function to the feature generation script as well for completeness (aka ease of copy-pasting)

File: src/psycopmlutils/feature_describer/feature_describer.py [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+        series (pd.Series): Series with data to describe.
+        predictor_dict (Dict[str, str]): Dictionary with predictor information.
+
+    Returns:
+        Dict: Dictionary with feature description.
+    """
+
+    d = {}
+
+    d["Raw values"] = predictor_dict["predictor_df"]
```
Comment (HLasse) [inline src/psycopmlutils/feature_describer/feature_describer.py:99]:
```suggestion
    d["Predictor df"] = predictor_dict["predictor_df"]
```

File: src/psycopmlutils/synth_data_generator/synth_data_generator.py [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+        # For all columns in df.columns if column is not in na_ignore_cols
+        for col in df.columns:
+            if col not in na_ignore_cols:
+                df[col] = df_[col]
+
+    return df
+
+
+def generate_data_columns(predictors, n_samples, df):
+    for col_name, col_props in predictors.items():
```
Comment (HLasse) [inline src/psycopmlutils/synth_data_generator/synth_data_generator.py:78]: Missing docstring and type hints

File: src/psycopmlutils/synth_data_generator/synth_data_generator.py [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+            "resolve_multiple": "max",
+            "fallback": 0,
+        },
+        {
+            "predictor_df": "hdl",
+            "lookbehind_days": 100,
+            "resolve_multiple": "max",
+            "fallback": "np.nan",
+        },
+    ]
```
Comment (HLasse) [inline src/psycopmlutils/synth_data_generator/synth_data_generator.py:163]: unused?

File: src/psycopmlutils/utils.py [hunk 5]
```diff
@@ -7,3 +7,25 @@
 SHARED_RESOURCES_PATH = Path(r"C:\shared_resources")
 FEATURE_SETS_PATH = SHARED_RESOURCES_PATH / "feature_sets"
 OUTCOME_DATA_PATH = SHARED_RESOURCES_PATH / "outcome_data"
+
+
+def generate_feature_colname(
```
Comment (HLasse) [inline src/psycopmlutils/utils.py:12]: 🙌

**Review by HLasse — COMMENTED** [review]
Mostly minor comments. I think we should write the `feature_describer` function in the same way as the deepchecks function, so one can pass in a directory and specify which splits to run it on. Will also make it more intuitive why one should pass a dir instead of a file

**Comment by MartinBernstorff** [comment]
Agreed on all counts! If you think the changes are satisfactory (specifically `feature_describer`), please approve so we can merge :-)

**Review by HLasse — APPROVED** [review]
All good!

**Pull request merged**